pkg/reconciler/kubernetes/initcontroller/initcontroller.go

```diff
@@ -41,7 +43,8 @@ type PayloadOptions struct {
 	ReadOnly bool
 }
 
-func (ctrl Controller) InitController(ctx context.Context, opts PayloadOptions) (mf.Manifest, string) {
+// InitController returns the component source manifest, operator release version and component release version
+func (ctrl Controller) InitController(ctx context.Context, opts PayloadOptions) (mf.Manifest, string, string) {
```

Comment:
How about
```suggestion
func NewControllerConfig (ctx context.Context, opts PayloadOptions) Controller {
```

then set manifest, operator version, payload version as struct fields.

In case of dashboard, we can call the function twice, one for readonly and then again for full access.

eg: for dashboard
```bash
		readonlyCtrlConfig := initcontroller.NewCtrlConfig(ctx, initcontroller.PayloadOptions{ReadOnly: true})
                fullaccessCtrlConfig := initcontroller.NewCtrlConfig(ctx, initcontroller.PayloadOptions{ReadOnly: true})
```